### PR TITLE
Praveen's WGLC comments (editorial)

### LINKED ISSUE #3919: Praveen's WGLC comments
From https://mailarchive.ietf.org/arch/msg/quic/_KeDg0RMQqP5epdYQdFb0M7MJ5s/, minus the question of limiting bursts, which is in Issue #3918

Section 2

" are not acknowledged, declared lost, or abandoned along with old keys." -->  are not acknowledged, declared lost, or discarded along with old keys."

Discarding is the term used in the rest of the doc notably in section 2.4. I see some reference to "abandon packets" in the transport draft as well. Would be good to use the same term for consistency everywhere.
 

Section 4.4. 

 "does not allow any acked packet to be reneged" --> "does not allow any acknowledged packet to be reneged"

Section 4.7

"QUIC uses a probe timeout (see Section 6.2)," --> "QUIC uses a Probe Timeout (PTO) (see Section 6.2),"

Section 5.2 "If the path's actual RTT increases, the min_rtt will not adapt to it, allowing future RTT samples that are smaller than the  new RTT be included in smoothed_rtt."

Would it safer to reset min_rtt after persistent congestion is declared. This is important so we can adapt to a consistent spike in RTT as a result of a path change.
 

Section 6.1.1

"Algorithms that increase the reordering threshold after spuriously detecting losses, such as TCP-NCR [RFC4653], have proven to be useful in TCP and are expected to at least as useful in QUIC."

It might be better to refer to RACK here which has an optional feature to increase reorder threshold in time. I don't know of any major implementations that do TCP-NCR but maybe I am mistaken.

Section 6.2.1

"time threshold Section 6.1.2" --> time threshold (Section 6.1.2)

“The life of a connection that is experiencing consecutive PTOs is limited by the endpoint's idle timeout.”

Is there any specification of recommended values for idle timeout? Any values lower than 9 seconds could adversely impact preserving of connectivity during VM live migration.

Section 6.2.4

"If the sender wants to elicit a faster acknowledgement on PTO, it can skip a packet number to eliminate the ack delay."

This adds some unnecessary complexity. Given that endpoint MAY send two full-sized datagrams, isn't that a better technique than skipping packet numbers?

Section 7.8

"The congestion period is calculated as the time between the oldest and newest lost packets: (3 - 0) = 3." --> "The congestion period is calculated as the sent time difference between the oldest and newest lost packets: (3 - 0) = 3."

It might be worth noting that persistent congestion can be declared without sending a PTO. Particularly because the example assumes PTOs. In the case where app kept on periodically sending new data and no ACKs came back for the threshold and then an ACK came back and declared all packets sent before it as lost.

Section 8.2

"Endpoints can use PADDING frames or bundle acknowledgments with other frames to reduce leaked information."
A reminder should be added here that PADDING counts towards in-flight so there is a possible impact on performance.

Section B.8

The function AreAllPacketsLost does not have any pseudocode.

### PR DESCRIPTION
**Pull request by janaiyengar**
Closes #3919.

### CODE REVIEW

File: draft-ietf-quic-recovery.md [hunk 1]
```diff
@@ -867,7 +867,9 @@ between the oldest and newest lost packets: (3 - 0) = 3.  The duration for
 persistent congestion is equal to: (1 * kPersistentCongestionThreshold) = 3.
 Because the threshold was reached and because none of the packets between the
 oldest and the newest packets are acknowledged, the network is considered to
-have experienced persistent congestion.
+have experienced persistent congestion. Note that this example shows the
+occurrence of PTOs, but that is not required for persistent congestion to be
+established.
```
Comment (martinthomson) [inline draft-ietf-quic-recovery.md:872]: Won't this conflict with the other change?

Comment (janaiyengar) [reply draft-ietf-quic-recovery.md:872]: Yup, I'll prefer that text when merging.

File: draft-ietf-quic-recovery.md [hunk 2]
```diff
@@ -957,8 +957,9 @@ sending rate by dropping packets, or alter send rate by changing ECN codepoints.
 
 Packets that carry only ACK frames can be heuristically identified by observing
 packet size.  Acknowledgement patterns may expose information about link
-characteristics or application behavior.  Endpoints can use PADDING frames or
-bundle acknowledgments with other frames to reduce leaked information.
+characteristics or application behavior.  To reduce leaked information,
+endpoints can use PADDING frames, at a potential cost to performance, or bundle
+acknowledgments with other frames.
```
Comment (ianswett) [inline draft-ietf-quic-recovery.md:962]:
```suggestion
characteristics or application behavior.  To reduce leaked information,
endpoints can bundle acknowledgments with other frames, or add
PADDING frames, at a potential cost to performance.
```

File: draft-ietf-quic-recovery.md [hunk 3]
```diff
@@ -436,12 +436,13 @@ The RECOMMENDED initial value for the packet reordering threshold
...
-Some networks may exhibit higher degrees of reordering, causing a sender to
-detect spurious losses.  Algorithms that increase the reordering threshold after
-spuriously detecting losses, such as TCP-NCR ({{?RFC4653}}), have proven to be
-useful in TCP and are expected to be at least as useful in QUIC.  Re-ordering
-could be more common with QUIC than TCP, because network elements cannot observe
-and fix the order of out-of-order packets.
+Some networks may exhibit higher degrees of packet reordering, causing a sender
+to detect spurious losses. Additionally, packet reordering could be more common
+with QUIC than TCP, because network elements that could observe and fix the
+order of reordered TCP packets cannot do it with QUIC. Algorithms that increase
```
Comment (martinthomson) [inline draft-ietf-quic-recovery.md:442]:
```suggestion
order of reordered TCP packets cannot do that for QUIC. Algorithms that increase
```

Comment (janaiyengar) [reply draft-ietf-quic-recovery.md:442]: Missed this. Fixed this in https://github.com/quicwg/base-drafts/commit/8abb9fa2354ae341637ce43479a707844aa4a7cf

**Pull request merged**